### ISSUE TITLE
require-return-type not reading object property names correctly

### ISSUE THREAD

**Issue by strutcode** [gajus/eslint-plugin-flowtype#395]
I was attempting to add an `excludeMatching` rule to `require-return-type` and noticed that it wasn't matching as expected. I started playing with the rule source and noticed that the `identifierName` was `undefined` which is why it wasn't matching.

Further inspection reveals that the `FunctionExpression` has a parent of type `Property` and a null id field, which causes the default selector to fail. I was able to fix it locally by changing this code:
```js
    const isProperty = functionNode.parent && functionNode.parent.type === 'ClassProperty';
```
To this:
```js
    const propertyNodes = ['Property', 'ClassProperty'];
    const isProperty = functionNode.parent && propertyNodes.includes(functionNode.parent.type);
```

I cloned the repo and tried to create a test case to reproduce the issue but couldn't. Maybe it's a version specific thing? I'm using `babel-eslint` as a parser with these options and these versions:

```js
    parserOptions: {
        parser: 'babel-eslint',
        ecmaVersion: 6,
        sourceType: 'module'
    },
```

```
babel-eslint@10.0.1
eslint-plugin-flowtype@3.4.2
eslint-plugin-vue@5.2.2
eslint@5.15.1
```
This is an example of offending code:
```js
export default {
  data() {
    return 42;
  }
}
```

**Comment by pnevyk** [comment]
I tried to add the test with code you presented and it indeed failed, while this code `function data() { return 42; }` passes. So this looks really like a bug.

What exactly didn't work for you when trying to create a test case? Did you try the full code example with `export default { ... }` or a simplified version of it?

**Comment by strutcode** [comment]
@pnevyk The exact assertion I added was:
```js
    {
      code: 'export default { foo() { return 42; } }',
      errors: [
        {
          message: 'Missing return type annotation.'
        }
      ],
      options: [
        'always',
        {
          annotateUndefined: 'always'
        }
      ]
    }
```
Which passes for me without changing the source but actually now that I think about it the trouble isn't that the rule fails, it's that the node name is undefined and so it can't be matched against the include and exclude options. I'll give testing it another shot and see if maybe I can submit a PR.

**Comment by strutcode** [comment]
Pull request created. Take a look and let me know if you think you can merge it as my team would like to use this rule with those options. Thanks!
